annotation-file-utilities/src/annotator/find/InheritedSymbolFinder.java

```diff
@@ -0,0 +1,211 @@
+package annotator.find;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.ElementFilter;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+import annotations.util.JVMNames;
+import annotator.Source;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.LinkedHashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Multimaps;
+
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.model.JavacElements;
+
+/**
+ * Provides static access to JVML descriptors of methods for which
+ *  implementations are inherited in a given class.
+ *
+ * @author dbro
+ */
+public class InheritedSymbolFinder {
+  private static Source source = null;
+
+  // class (flat) name -> method index
+  private static Map<String, Multimap<String, ExecutableElement>> cache =
+      new TreeMap<String, Multimap<String, ExecutableElement>>();
+
+  // keys: class (flat) name
+  // values: multimap of method (simple) name to matching method elems
+  private static Map<String, Multimap<String, ExecutableElement>> cash =
+      new HashMap<String, Multimap<String, ExecutableElement>>();
+
+  // initialized once and for all
+  private static Multimap<String, ExecutableElement> objectMethods = null;
+
+  private InheritedSymbolFinder() {}
+
+  // Needed for obtaining instances of utility classes Types and Elements.
+  // TODO: make methods non-static and pass source to constructor instead.
+  public static void setSource(Source source) {
+    if (InheritedSymbolFinder.source != source) {
+      InheritedSymbolFinder.source = source;
+      cache.clear();
+      cash.clear();
+      if (source != null) {
+        // initialize objectMethods
+        if (objectMethods == null) {
+          TypeElement o =
+              source.getElementUtils().getTypeElement("java.lang.Object");
+          objectMethods =
+              LinkedHashMultimap.<String, ExecutableElement>create();
+          for (Element e : o.getEnclosedElements()) {
+            if (e.getKind() == ElementKind.METHOD) {
+              String key = keyFor((Symbol.MethodSymbol) e);
+              objectMethods.put(key, (ExecutableElement) e);
+            }
+          }
+        }
+        cash.put("java.lang.Object", objectMethods);
+      }
+    }
+  }
+
+  /**
+   * Finds whether a method matching a JVML descriptor has an inherited
+   *  implementation in a given class.
```

Comment:
Don't put extra indentation on the second line.  You did this in some other places, too, so please fix them all.